app/models/slide.rb

```diff
@@ -68,8 +68,13 @@ def page_list_url
   end
 
   def page_list
-    url = self.page_list_url
-    get_json(url)
+    len = num_of_pages.abs.to_s.length
+    result = []
+    for i in 1..num_of_pages
```

Comment:
__[rubocop]__
* Style/For: Prefer `each` over `for`. [:ref](https://github.com/bbatsov/ruby-style-guide#no-for-loops)